spec/unit/hanami/cli/commands/app/generate/part_spec.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+require "hanami"
+require "ostruct"
+
+RSpec.describe Hanami::CLI::Commands::App::Generate::Part, :app do
+  subject { described_class.new(fs: fs, inflector: inflector, generator: generator) }
+
+  let(:out) { StringIO.new }
+  let(:fs) { Hanami::CLI::Files.new(memory: true, out: out) }
+  let(:inflector) { Dry::Inflector.new }
+  let(:generator) { Hanami::CLI::Generators::App::Part.new(fs: fs, inflector: inflector) }
+  let(:app) { Hanami.app.namespace }
+  let(:dir) { inflector.underscore(app) }
+
+  def output
+    out.rewind && out.read.chomp
+  end
+
+  context "generating for app" do
+    it "generates a part in a top-level namespace" do
+      within_application_directory do
+        subject.call(name: "user")
+
+        # part
+        part = <<~EXPECTED
+          # auto_register: false
+          # frozen_string_literal: true
+
+          module Test
+            module Views
+              module Parts
+                class User < Test::Part
+                end
+              end
+            end
+          end
+        EXPECTED
+
+        expect(fs.read("app/views/parts/user.rb")).to eq(part)
+        expect(output).to include("Created app/views/parts/user.rb")
+      end
+    end
+  end
+
+  context "generating for a slice" do
+    it "generates a view in a top-level namespace" do
+      within_application_directory do
+        fs.mkdir("slices/main")
+        subject.call(name: "user", slice: "main")
+
+        # view
+        view_file = <<~EXPECTED
```

Comment:
```suggestion
        part_file = <<~EXPECTED
```

(Might be worth reviewing the rest of this test for any other copy/paste inconsistencies like this)